File: python/test/test_trtc.py
import ThrustRTC as trtc
import numpy as np



# interface with numpy
harr = np.array([1.0, 2.0, 3.0, 4.0, 5.0], dtype='float32')
darr = trtc.device_vector_from_numpy(harr)
print(darr.to_host())

# C data type
print(darr.name_view_cls())

harr2 = np.array([6,7,8,9,10], dtype='int32')
darr2 = trtc.device_vector_from_numpy(harr2)

# kernel with auto parameters, launched twice with different types
kernel = trtc.Kernel(['arr_in', 'arr_out', 'k'],
	'''
	size_t idx = blockIdx.x * blockDim.x + threadIdx.x;
	if (idx >= arr_in.size()) return;
	arr_out[idx] = arr_in[idx]*k;
	''')

darr_out = trtc.device_vector('float', 5)
kernel.launch(1,128, [darr, darr_out, trtc.DVFloat(10.0)])
print (darr_out.to_host())

darr_out = trtc.device_vector('int32_t', 5)
kernel.launch(1,128, [darr2, darr_out, trtc.DVInt32(5)])
print (darr_out.to_host())

# create a vector from python list with C type specified
darr3 = trtc.device_vector_from_list([3.0, 5.0, 7.0, 9.0 , 11.0], 'float')
print(darr3.to_host())

File: java/test/test_trtc.java
import java.util.*;
import JThrustRTC.*;

public class test_trtc 
{
	public static void main(String[] args) 
	{

	    Kernel ker = new Kernel(new String[]{ "arr_in", "arr_out", "k" }, 
	   	String.join("",
	  	"    size_t idx = blockIdx.x * blockDim.x + threadIdx.x;\n",
	    "    if (idx >= arr_in.size()) return;\n",
	    "    arr_out[idx] = arr_in[idx]*k;\n"));

	    DVVector dvec_in_f = new DVVector(new float[] { 1.0f, 2.0f, 3.0f, 4.0f, 5.0f });
	    DVVector dvec_out_f = new DVVector("float", 5);
	    DVFloat k1 = new DVFloat(10.0f);
	    DeviceViewable[] args_f = new DeviceViewable[] { dvec_in_f, dvec_out_f, k1 };
	    ker.launch(1, 128, args_f, 0);
	    System.out.println(Arrays.toString((float[])dvec_out_f.to_host()));

	    DVVector dvec_in_i = new DVVector(new int[] { 6, 7, 8, 9, 10 });
	    DVVector dvec_out_i = new DVVector("int32_t", 5);
	    DVInt32 k2 = new DVInt32(5);
	    DeviceViewable[] args_i = new DeviceViewable[] { dvec_in_i, dvec_out_i, k2 };
	    ker.launch(1, 128, args_i, 0);
	    System.out.println(Arrays.toString((int[])dvec_out_i.to_host()));


	}
}
